app/views/base.py

```diff
@@ -210,6 +210,9 @@ def build_absolute_path(self, path=None):
             path (str, optional): A path beginning with a slash (may include a
                 query string), e.g., '/abc?x=y'. If the path argument is not
                 specified or is None, the current request's path will be used.
+            repo (str, optional): A repo ID. If specified, the path will be
+                considered relative to the repo's route. If this is specified,
+                path must also be specified.
```

Comment:
currently if the user specifies repo but not path, the first branch of the if-statement will execute, which may be unexpected. maybe add an assertion that path must be present if repo is present?

Reply:
Done.